packages/typespec-vscode/src/extension.ts

```diff
@@ -116,48 +198,55 @@ async function resolveTypeSpecServer(context: ExtensionContext): Promise<Executa
 
   // In production, first try VS Code configuration, which allows a global machine
   // location that is not on PATH, or a workspace-specific installation.
-  let serverPath: string | undefined = workspace.getConfiguration().get("typespec.tsp-server.path");
-  if (serverPath && typeof serverPath !== "string") {
-    throw new Error("VS Code configuration option 'typespec.tsp-server.path' must be a string");
+  let targetPath: string | undefined = workspace.getConfiguration().get(`typespec.${cmdName}.path`);
+  if (targetPath && typeof targetPath !== "string") {
+    throw new Error(`VS Code configuration option 'typespec.${cmdName}.path' must be a string`);
   }
   const workspaceFolder = workspace.workspaceFolders?.[0]?.uri?.fsPath ?? "";
 
-  // Default to tsp-server on PATH, which would come from `npm install -g
+  // Default to tsp/tsp-server on PATH, which would come from `npm install -g
   // @typespec/compiler` in a vanilla setup.
-  if (serverPath) {
-    outputChannel.debug(`Server path loaded from VS Code configuration: ${serverPath}`);
+  if (targetPath) {
+    logger.debug(`Server path loaded from VS Code configuration: ${targetPath}`);
   } else {
-    serverPath = await resolveLocalCompiler(workspaceFolder);
+    targetPath = await resolveLocalCompiler(workspaceFolder);
   }
-  if (!serverPath) {
-    const executable = process.platform === "win32" ? "tsp-server.cmd" : "tsp-server";
-    outputChannel.debug(`Can't resolve server path, try to use default value ${executable}.`);
+  if (!targetPath) {
+    const executable = process.platform === "win32" ? `${cmdName}.cmd` : cmdName;
+    logger.debug(`Can't resolve ${cmdName} path, try to use default value ${executable}.`);
     return { command: executable, args, options };
   }
   const variableResolver = new VSCodeVariableResolver({
     workspaceFolder,
     workspaceRoot: workspaceFolder, // workspaceRoot is deprecated but we still support it for backwards compatibility.
   });
 
-  serverPath = variableResolver.resolve(serverPath);
-  outputChannel.debug(`Server path expanded to: ${serverPath}`);
+  targetPath = variableResolver.resolve(targetPath);
+  logger.debug(`Server path expanded to: ${targetPath}`);
 
-  if (!serverPath.endsWith(".js")) {
-    // Allow path to tsp-server.cmd to be passed.
-    if (await isFile(serverPath)) {
+  if (!targetPath.endsWith(".js")) {
+    // Allow path to tsp.cmd or tsp-server.cmd to be passed.
+    if (await isFile(targetPath)) {
       const command =
-        process.platform === "win32" && !serverPath.endsWith(".cmd")
-          ? `${serverPath}.cmd`
-          : "tsp-server";
+        process.platform === "win32" && !targetPath.endsWith(".cmd")
+          ? `${targetPath}.cmd`
+          : cmdName;
 
```

Comment:
@timotheeguerin, in the original code (line151), it feels like "serverPath" should be used instead of "tsp-server". would like to confirm whether it's a bug or by design for some specific scenario? thanks.